### PR TITLE
jitsi-meet: stable/jitsi-meet_5390 -> stable/jitsi-meet_5963

### PR DESCRIPTION
**Pull request by illustris**
###### Motivation for this change
https://github.com/NixOS/nixpkgs/pull/120583 is out of date and appears to have stalled. Motivation remains the same, "Jitsi has gotten more functionality visible in the UI in the most recent version over the one that is currently in nixpkgs, there is also a [variety of fixes](https://github.com/jitsi/jitsi-meet-release-notes/blob/master/CHANGELOG-WEB.md)."

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

This PR is a WIP. The newer versions of jitsi-meet and jitsi-videobridge work fine, but updating Jicofo breaks Jitsi.

### CODE REVIEW

**Comment by illustris** [comment]
Tagging maintainers
@mmilata @petabyteboy @ryantm

Does anyone know why upgrading jicofo causes issues? It might also be just on my setup, is anyone able to reproduce this issue?

**Comment by illustris** [comment]
https://github.com/NixOS/nixpkgs/pull/127236/commits/0d245614871bec72af16239464cff622671d73fc should account for the changes made in https://github.com/jitsi/jitsi-meet/commit/b6f7f8fba7ba81c5a0ab5477a1d103b7279bf38f.

There is one last issue though. The [prosody-plugins](https://github.com/jitsi/jitsi-meet/tree/stable/jitsi-meet_5963/resources/prosody-plugins) folder does not exist in the Jitsi web deb file or "https://download.jitsi.org/jitsi-meet/src/jitsi-meet-${version}.tar.bz2". The plugins folder must be included in the jitsi-meet service definition as `services.prosody.extraPluginPaths = [ "/path/to/prosody-plugins" ];`. I tested this in my deployment by just copying the folder locally and adding `extraPluginPaths = [ ./prosody-plugins ];`. Should this folder just be fetched from github, or is there a better way?

**Comment by illustris** [comment]
Tested, works. @mmilata @petabyteboy @ryantm please review.

File: pkgs/misc/jitsi-meet-prosody/default.nix [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+let
+  pname = "jitsi-meet-prosody";
+  version = "1.0.5056";
+  src = fetchurl {
+    url = "https://download.jitsi.org/stable/${pname}_${version}-1_all.deb";
+    sha256 = "06qxa9h2ry92xrk2jklp76nv3sl8nvykdvsqmhn33lz6q6vmw2xr";
+  };
+in
+stdenv.mkDerivation {
+  inherit pname version src;
```
Comment (SuperSandro2000) [inline pkgs/misc/jitsi-meet-prosody/default.nix:12]:
```suggestion
stdenv.mkDerivation rec {
  pname = "jitsi-meet-prosody";
  version = "1.0.5056";

  src = fetchurl {
    url = "https://download.jitsi.org/stable/${pname}_${version}-1_all.deb";
    sha256 = "06qxa9h2ry92xrk2jklp76nv3sl8nvykdvsqmhn33lz6q6vmw2xr";
  };
```
Don't be afraid of rec.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/misc/jitsi-meet-prosody/default.nix (hunk 1 above)
Comment (illustris) [reply pkgs/misc/jitsi-meet-prosody/default.nix]: I keep forgetting this is a thing... Updated

**Comment by petabyteboy** [comment]
Works fine in manual testing, but the NixOS tests time out?

**Comment by illustris** [comment]
@petabyteboy I've updated the tests

**Comment by petabyteboy** [comment]
In local testing the test runs out of memory, but adding `virtualisation.memorySize = 512;` to the server block fixes that.

**Comment by illustris** [comment]
Huh... yes, looks like it fails sometimes. It worked the first time, failed the second time and worked again the third time. Updated test configs.

**Comment by petabyteboy** [comment]
Please add a `nixos/jitsi-meet: ` prefix to the commit message of 0d245614871bec72af16239464cff622671d73fc

**Comment by illustris** [comment]
Updated commit message prefixes

**Comment by petabyteboy** [comment]
Thanks a lot!

**Pull request merged**